source/source.go

```diff
@@ -102,6 +102,15 @@ func getProviderSpecificAnnotations(annotations map[string]string) endpoint.Prov
 			Value: "true",
 		})
 	}
+	for k, v := range annotations {
```

Comment:
you could do it similar what we did with CloudflareProxiedKey some lines below, would this be enough?

Reply:
This logic, I see if the implementation of CloudflareProxiedKey meets the requirements.